File: internal/domain/category.go
package domain

import (
	"time"
)

// Category represents an item category
type Category struct {
	ID          int64   `json:"id"`
	Name        string  `json:"name"`
	Slug        string  `json:"slug"`
	Description *string `json:"description,omitempty"`
	ParentID    *int64  `json:"parent_id,omitempty"`
	Icon        *string `json:"icon,omitempty"`

	// Loan settings
	DefaultInterestRate float64  `json:"default_interest_rate"`
	MinLoanAmount       *float64 `json:"min_loan_amount,omitempty"`
	MaxLoanAmount       *float64 `json:"max_loan_amount,omitempty"`
	LoanToValueRatio    float64  `json:"loan_to_value_ratio"`

	// Display
	SortOrder int  `json:"sort_order"`
	IsActive  bool `json:"is_active"`

	// Timestamps
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	// Relations
	Parent   *Category   `json:"parent,omitempty"`
	Children []*Category `json:"children,omitempty"`
}

// TableName returns the database table name
func (Category) TableName() string {
	return "categories"
}

// IsRoot checks if the category is a root category (no parent)
func (c *Category) IsRoot() bool {
	return c.ParentID == nil
}

File: web/src/types/category.ts
// Category types - mirrors internal/domain/category.go

export interface Category {
  id: number
  name: string
  slug: string
  description?: string
  parent_id?: number
  icon?: string
  default_interest_rate: number
  min_loan_amount?: number
  max_loan_amount?: number
  loan_to_value_ratio: number
  sort_order: number
  is_active: boolean
  created_at: string
  updated_at: string
  parent?: Category
  children?: Category[]
}

export interface CreateCategoryInput {
  name: string
  description?: string
  parent_id?: number
  icon?: string
  default_interest_rate?: number
  min_loan_amount?: number
  max_loan_amount?: number
  loan_to_value_ratio?: number
  sort_order?: number
}

export interface UpdateCategoryInput {
  name?: string
  description?: string
  parent_id?: number
  icon?: string
  default_interest_rate?: number
  min_loan_amount?: number
  max_loan_amount?: number
  loan_to_value_ratio?: number
  sort_order?: number
  is_active?: boolean
}

export interface CategoryListParams {
  page?: number
  per_page?: number
  order_by?: string
  order?: 'asc' | 'desc'
  search?: string
  parent_id?: number
  is_active?: boolean
}
